app/assets/javascripts/discourse/app/lib/screen-track.js

```diff
@@ -85,6 +89,99 @@ export default class {
     this._anonCallback = cb;
   }
 
+  consolidateTimings(timings, topicTime, topicId) {
+    this._consolidatedTimings = this._consolidatedTimings || [];
+
+    let found = this._consolidatedTimings.find((elem) => elem[2] === topicId);
+
+    if (found) {
+      this._consolidatedTimings.removeObject(found);
+
+      const oldTimings = found[0];
+      Object.keys(oldTimings).forEach((id) => {
+        if (timings[id]) {
+          oldTimings[id] += timings[id];
+        }
+      });
+      found[1] += topicTime;
+      found[0] = Object.assign({}, timings, found[0]);
+    }
+
+    found = found || [timings, topicTime, topicId];
+    this._consolidatedTimings.unshift(found);
+
+    return this._consolidatedTimings;
+  }
+
+  sendNextConsolidatedTiming() {
+    if (!this._consolidatedTimings || this._consolidatedTimings.length === 0) {
+      return Promise.resolve();
+    }
+
+    if (this._inProgress) {
+      return Promise.resolve();
+    }
+
+    if (
+      this._blockSendingToServerTill &&
+      this._blockSendingToServerTill > Date.now()
+    ) {
+      return Promise.resolve();
+    }
+
+    this._ajaxFailures = this._ajaxFailures || 0;
+
+    const item = this._consolidatedTimings.shift();
+    const timings = item[0];
```

Comment:
FWIW you can rewrite this like it:

```
const [timings, topicTime, topicid] = item;
```

Reply:
Or if you use an object as I suggested above: `const { timings, topictime, topicId } = item;`

Reply:
done, moved to an object